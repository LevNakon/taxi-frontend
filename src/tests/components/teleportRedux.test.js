import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Teleport from '../../components/teleport';

describe('Teleport Redux --- test render correctly component', () => {
    const initialState = {
        userState: {
            user: {
                firstName: 'test firstName',
                lastName: 'test lastName',
                email: 'test email',
                password: 'test password',
                gender: 'MALE',
                birthdayDate: 'test birthdayDate',
                avatarUrl: 'test avatarUrl',
                mobileNumber: 'test mobileNumber',
                homeAddress: 'test homeAddress',
                workAddress: 'test workAddress',
                driverId: null
            },
            isChecked: false
        }
    };
    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<Teleport store={store} />);
    });

    it('render correctly component Teleport', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('check props matches with initialState', () => {
        expect(wrapper.dive().prop('user')).toEqual(initialState.userState.user);
        expect(wrapper.dive().prop('isChecked')).toEqual(initialState.userState.isChecked);
    });
});