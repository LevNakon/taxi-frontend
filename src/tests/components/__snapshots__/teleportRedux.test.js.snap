// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teleport Redux --- test render correctly component render correctly component Teleport 1`] = `
<ContextProvider
  value={null}
>
  <withRouter(Teleport)
    checkerChange={[Function]}
    isChecked={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    tripCreateWatcher={[Function]}
    user={
      Object {
        "avatarUrl": "test avatarUrl",
        "birthdayDate": "test birthdayDate",
        "driverId": null,
        "email": "test email",
        "firstName": "test firstName",
        "gender": "MALE",
        "homeAddress": "test homeAddress",
        "lastName": "test lastName",
        "mobileNumber": "test mobileNumber",
        "password": "test password",
        "workAddress": "test workAddress",
      }
    }
    userGetWatcher={[Function]}
  />
</ContextProvider>
`;
